src/extension.ts

```diff
@@ -365,49 +365,55 @@ class DeviceTreeItem extends vscode.TreeItem {
     }
     
     async captureScreenshot() {
-        const statusBarMessage = new StatusBarProgressionMessage("Attempting to capture screenshot...");
-
-        const handleCaptureError = e => {
-            vscode.window.showErrorMessage("Error capturing screenshot: " + (e.message || e)); 
-            statusBarMessage.finish();
-        }
-
-        try {
-            const screenshotDirectory = await getSharedTempDir('ev3dev-screenshots');
-            const screenshotBaseName = `ev3dev-${sanitizedDateString()}.png`;
-            const screenshotFile = `${screenshotDirectory}/${screenshotBaseName}`;
-
-            const conn = await this.device.exec('fbgrab -');
-            const writeStream = fs.createWriteStream(screenshotFile);
-
-            conn.on('error', (e: Error) => {
-                writeStream.removeAllListeners('finish');
-                handleCaptureError(e);
-            });
-
-            writeStream.on('open', () => {
-                conn.stdout.pipe(writeStream);
-            });
-            
-            writeStream.on('error', (e: Error) => {
-                vscode.window.showErrorMessage("Error saving screenshot: " + e.message);
-                statusBarMessage.finish();
-            });
-
-            writeStream.on('finish', async () => {
-                const pngHeader = [ 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A ];
-                if (await verifyFileHeader(screenshotFile, pngHeader)) {
-                    statusBarMessage.finish(`Screenshot "${screenshotBaseName}" successfully captured`);
-                    vscode.commands.executeCommand('vscode.open', vscode.Uri.file(screenshotFile), vscode.ViewColumn.Two);
+        vscode.window.withProgress({
+            location: vscode.ProgressLocation.Window,
+            title: "Attempting to capture screenshot..."

```

Comment:
While we are touching this, we could make it more succinct. "Capturing screenshot..." gets the message across. It happens so fast, most people won't have time to read this anyway.